src/Bicep.Core/Analyzers/Linter/LinterRuleBase.cs

```diff
@@ -0,0 +1,161 @@
+        {
+            this.AnalyzerName = LinterAnalyzer.AnalyzerName;
+            this.Code = code;
+            this.Description = description;
+            this.DocumentationUri = docUri;
+            this.DiagnosticLevel = diagnosticLevel;
+            this.DiagnosticLabel = diagnosticLabel;
+        }
+
+        private IConfigurationRoot? Config;
+        public string AnalyzerName { get; }
+
+        public string Code { get; }
+        public readonly string RuleConfigSection = $"{LinterAnalyzer.SettingsRoot}:{LinterAnalyzer.AnalyzerName}:rules";
+        public Diagnostics.DiagnosticLevel DiagnosticLevel { get; private set; }
+        public string Description { get; }
+        public string DocumentationUri { get; }
+        public Diagnostics.DiagnosticLabel? DiagnosticLabel { get; }
+
+
+        /// <summary>
+        /// Override to implement detailed message for rule
+        /// </summary>
+        /// <param name="values"></param>
+        /// <returns></returns>
+        protected virtual string FormatMessage(params object[] values) => this.Description;
+
+        public virtual void Configure(IConfigurationRoot config)
+        {
+            this.Config = config;
+            var configDiagLevel = GetConfiguration("level", this.DiagnosticLevel.ToString());
+            if (DiagnosticLevel.TryParse<DiagnosticLevel>(configDiagLevel, true, out var lvl))
+            {
+                this.DiagnosticLevel = lvl;
+            }
+        }
+
+        /// <summary>
+        /// Gets a message using the supplied parameter values (if any).
+        /// Otherwise returns the rule description
+        /// </summary>
+        /// <param name="values"></param>
+        /// <returns></returns>
+        public string GetMessage(params object[] values)
+        {
+            return (values.Any() ? FormatMessage(values) : this.Description)
+                + "\n" // this is used for cross-platform compatibility
+                + string.Format(CoreResources.SeeDocLinkFormat, this.DocumentationUri);
+        }
+
+        public IEnumerable<IBicepAnalyzerDiagnostic> Analyze(SemanticModel model)
+        {
+            try
+            {
+                // Expand the iteration immediately or the try/catch won't catch exceptions occuring during the rule analysis
+                // TODO: We need exception handling further up the tree in order to handle external linters.
+                return AnalyzeInternal(model).ToArray();
+            }
+            catch (Exception ex)
+            {
+                return new[]{ new AnalyzerDiagnostic(this.AnalyzerName,
+                                                    new TextSpan(0, 0),
+                                                    DiagnosticLevel.Warning,
+                                                    LinterAnalyzer.FailedRuleCode,
+                                                    string.Format(CoreResources.LinterRuleExceptionMessageFormat,this.AnalyzerName, ex.Message),
```

Comment:
Is this really something we want to expose to the end-user? Feels like it could be a little cryptic and unactionable.

Reply:
@anthony-c-martin It's not expected to happen, however do we have an established pattern of where to send this information for rule developers?

Reply:
We have to either send telemetry or show an error to the user so they can complain to the rule creator or both.

The assumption from Marcin was that this is better than showing a "fatal error" that blocks the compilation.

Reply:
Maybe a pop up box asking if they want to report it. Making sure they can dismiss it so it isn't continually popping up.